### PR TITLE
gh-103977: compile re expressions in platform.py only if required

### PR DESCRIPTION
**Pull request by eendebakpt**
In `platform.py` there are several  regular expressions compiled during import that might not be used at all. The regular expressions are only called once at most, since they are called from within cached methods. We can eliminate the cost of compilation by moving the regular expressions into the methods.

The import time of `platform.py` (measured with https://github.com/sympy/sympy/blob/master/bin/test_import, adapted for platform.py):

```
Main: The speed of "import platform" is: 7.567131 +- 0.208541 [ms]
PR: The speed of "import platform" is: 6.513338 +- 0.404813 [ms]
```

Notes:

* All regular expressions modified are within cached methods. The only exceptions is are `_libc_search` and `component_re` which are used in the method `libc_ver` (not cached). The `libc_ver` is called from `platform` which is cached.
If we expect `libc_ver` to be called many times, it should be cached as well, or the `_libc_search` regular expression moved back as a module variable.
* A couple of ms of the import time is spend in importing other modules (such as `collections`)

@hauntsaninja This is a followup of #102492

* Issue: gh-103977

### CODE REVIEW

**Comment by JelleZijlstra** [comment]
Can you run `make patchcheck` to fix the whitespace issue?

**Comment by TeamSpen210** [comment]
Since they’re now all local variables, it might be nice to remove the underscore prefixes.

File: Lib/test/coding20731.py [hunk 1]
```diff
@@ -1,4 +1 @@
 #coding:latin1
```
Comment (JelleZijlstra) [inline Lib/test/coding20731.py]: I don't think this one should be changed.

Comment (eendebakpt) [reply Lib/test/coding20731.py]: You are right. The make patchcheck gave some problems for me, so I executed some scripts manually. I guess that triggered the changes. I will revert them

File: Misc/NEWS.d/next/Library/2023-04-28-19-08-50.gh-issue-103977.msF70A.rst [hunk 2]
```diff
@@ -0,0 +1 @@
+Improve import time of platform module.
```
Comment (JelleZijlstra) [inline Misc/NEWS.d/next/Library/2023-04-28-19-08-50.gh-issue-103977.msF70A.rst]:
```suggestion
Improve import time of :mod:`platform` module.
```

Comment (hauntsaninja) [reply Misc/NEWS.d/next/Library/2023-04-28-19-08-50.gh-issue-103977.msF70A.rst]: Should I ship a bot for this?

Screenshot 2023-04-29 at 3 21 09 PM

**Review by JelleZijlstra — APPROVED** [review]

**Comment by JelleZijlstra** [comment]
Seems unlikely that this would cause a segfault in test_threading.

**Comment by hugovk** [comment]
And backport to 3.11?

**Comment by JelleZijlstra** [comment]
I don't think so; we don't backport perf improvements.

**Pull request merged**